### ISSUE TITLE
[ML] Regression job with excessive logging x = NaN 

### ISSUE THREAD

**Issue by sophiec20** [elastic/ml-cpp#729]
Found in 7.5.0-SNAPSHOT `    "build_hash" : "ecb20ebc6cb155be37ecdc68ca8a178531744758",
    "build_date" : "2019-10-10T09:21:05.340310Z",`

For a job using the kibana sample dataset for flights, the following config caused excessive log errors. The job also stayed at high 90's% for quite a while. I stopped it before finding out if it would finish.

```
PUT _ml/data_frame/analytics/flights-delays01 
{
  "source": {
    "index": [ "kibana_sample_data_flights" ],
    "query": {
      "range": { "DistanceKilometers": { "gt": 0 }}
    }
  },
  "dest": { "index": "df-flights-delays01" },
  "analysis": {
    "regression": {
      "dependent_variable": "FlightDelayMin",
      "training_percent": 90
    }
  },
  "analyzed_fields": {
    "includes": [],
    "excludes": [
      "Cancelled",
      "FlightDelay",
      "FlightDelayType"
    ]
  },
  "model_memory_limit": "100mb"
}
```

Admittedly this is not the best curated data for experimenting with regression, but the  log is excessive:

```
[2019-10-10T17:40:27,551][ERROR][o.e.x.m.p.l.CppLogMessageHandler] [node3] [flights-delays01] [data_frame_analyzer/5403] [CTools.cc@147] x = NaN, distribution = N5boost4math19normal_distributionIdNS0_8policies6policyINS2_14overflow_errorILNS2_17error_policy_typeE3EEENS2_14default_policyES7_S7_S7_S7_S7_S7_S7_S7_S7_S7_S7_EEEE
[2019-10-10T17:40:27,552][ERROR][o.e.x.m.p.l.CppLogMessageHandler] [node3] [flights-delays01] [data_frame_analyzer/5403] [CTools.cc@117] x = NaN, distribution = N5boost4math19normal_distributionIdNS0_8policies6policyINS2_14overflow_errorILNS2_17error_policy_typeE3EEENS2_14default_policyES7_S7_S7_S7_S7_S7_S7_S7_S7_S7_S7_EEEE
[2019-10-10T17:40:27,552][ERROR][o.e.x.m.p.l.CppLogMessageHandler] [node3] [flights-delays01] [data_frame_analyzer/5403] [CTools.cc@147] x = NaN, distribution = N5boost4math19normal_distributionIdNS0_8policies6policyINS2_14overflow_errorILNS2_17error_policy_typeE3EEENS2_14default_policyES7_S7_S7_S7_S7_S7_S7_S7_S7_S7_S7_EEEE
[2019-10-10T17:40:27,552][ERROR][o.e.x.m.p.l.CppLogMessageHandler] [node3] [flights-delays01] [data_frame_analyzer/5403] [CTools.cc@117] x = NaN, distribution = N5boost4math19normal_distributionIdNS0_8policies6policyINS2_14overflow_errorILNS2_17error_policy_typeE3EEENS2_14default_policyES7_S7_S7_S7_S7_S7_S7_S7_S7_S7_S7_EEEE
[2019-10-10T17:40:27,552][ERROR][o.e.x.m.p.l.CppLogMessageHandler] [node3] [flights-delays01] [data_frame_analyzer/5403] [CTools.cc@147] x = NaN, distribution = N5boost4math19normal_distributionIdNS0_8policies6policyINS2_14overflow_errorILNS2_17error_policy_typeE3EEENS2_14default_policyES7_S7_S7_S7_S7_S7_S7_S7_S7_S7_S7_EEEE
[2019-10-10T17:40:27,552][ERROR][o.e.x.m.p.l.CppLogMessageHandler] [node3] [flights-delays01] [data_frame_analyzer/5403] [CTools.cc@117] x = NaN, distribution = N5boost4math19normal_distributionIdNS0_8policies6policyINS2_14overflow_errorILNS2_17error_policy_typeE3EEENS2_14default_policyES7_S7_S7_S7_S7_S7_S7_S7_S7_S7_S7_EEEE
[2019-10-10T17:40:27,552][ERROR][o.e.x.m.p.l.CppLogMessageHandler] [node3] [flights-delays01] [data_frame_analyzer/5403] [CTools.cc@147] x = NaN, distribution = N5boost4math19normal_distributionIdNS0_8policies6policyINS2_14overflow_errorILNS2_17error_policy_typeE3EEENS2_14default_policyES7_S7_S7_S7_S7_S7_S7_S7_S7_S7_S7_EEEE
[2019-10-10T17:40:27,552][ERROR][o.e.x.m.p.l.CppLogMessageHandler] [node3] [flights-delays01] [data_frame_analyzer/5403] [CTools.cc@117] x = NaN, distribution = N5boost4math19normal_distributionIdNS0_8policies6policyINS2_14overflow_errorILNS2_17error_policy_typeE3EEENS2_14default_policyES7_S7_S7_S7_S7_S7_S7_S7_S7_S7_S7_EEEE
[2019-10-10T17:40:27,552][ERROR][o.e.x.m.p.l.CppLogMessageHandler] [node3] [flights-delays01] [data_frame_analyzer/5403] [CTools.cc@147] x = NaN, distribution = N5boost4math19normal_distributionIdNS0_8policies6policyINS2_14overflow_errorILNS2_17error_policy_typeE3EEENS2_14default_policyES7_S7_S7_S7_S7_S7_S7_S7_S7_S7_S7_EEEE
[2019-10-10T17:40:27,552][ERROR][o.e.x.m.p.l.CppLogMessageHandler] [node3] [flights-delays01] [data_frame_analyzer/5403] [CTools.cc@117] x = NaN, distribution = N5boost4math19normal_distributionIdNS0_8policies6policyINS2_14overflow_errorILNS2_17error_policy_typeE3EEENS2_14default_policyES7_S7_S7_S7_S7_S7_S7_S7_S7_S7_S7_EEEE
[2019-10-10T17:40:27,552][ERROR][o.e.x.m.p.l.CppLogMessageHandler] [node3] [flights-delays01] [data_frame_analyzer/5403] [CTools.cc@147] x = NaN, distribution = N5boost4math19normal_distributionIdNS0_8policies6policyINS2_14overflow_errorILNS2_17error_policy_typeE3EEENS2_14default_policyES7_S7_S7_S7_S7_S7_S7_S7_S7_S7_S7_EEEE```

**Comment by tveasey** [comment]
Log throttling is addressed by #1615. It would be good to understand where the NaNs are coming from in this case.

**Comment by tveasey** [comment]
I tried this a couple of times and failed to reproduce (since this isn't seeded different runs can produce different results). I also audited the code and don't think this functionality is called from regression anymore. I'm therefore going to close this now we have throttling in case of excessive error logging.
